can't build optimised `tutorial/httpd` on macOS

macOS M1 installed from the Homebrew formula

```

λ gerbil -v
Gerbil v0.18.1-12-gf7d8efcf on Gambit v4.9.5-78-g8b18ab69
λ gerbil build -O
... build in current directory
... compile simpled
... compile exe simpled -> /Users/jan/dev/scheme/gerbil/src/tutorial/httpd/.gerbil/bin/simpled
/Users/jan/dev/scheme/gerbil/src/tutorial/httpd/.gerbil/bin/simpled__exe.c:
simpled__exe.c:30389:10: fatal error: openssl/crypto.h: No such file or directory
30389 | #include <openssl/crypto.h>
      |          ^~~~~~~~~~~~~~~~~~
compilation terminated.
*** ERROR IN ##main -- target compilation or link failed while compiling "/Users/jan/dev/scheme/gerbil/src/tutorial/httpd/.gerbil/bin/simpled__exe.c"
*** ERROR IN std/misc/concurrent-plan#perform-plan/threads__% -- Build Failure at (exe simpled)
```

```
λ md5sum /opt/homebrew/include/openssl/crypto.h
cd75ae8274a4de9c0be93d2411012392  /opt/homebrew/include/openssl/crypto.h
```

--- a/src/tutorial/httpd/build.ss
+++ b/src/tutorial/httpd/build.ss
@@ -2,6 +2,15 @@
 ;; -*- Gerbil -*-
 
 (import :std/build-script)
+(import :std/make)
 
 (defbuild-script
-  '((exe: "simpled")))
+  (cond-expand
+    (darwin
+     `((exe: "simpled"
+        "-cc-options" ,(cppflags "openssl" "")
+        "-cc-options" "-Wno-implicit-function-declaration"
+        "-cc-options" "-Wno-deprecated-declarations"
+        "-ld-options" ,(ldflags "openssl" ""))))
+    (else
+     '((exe: "simpled")))))

